api/client/swarm/init.go

```diff
@@ -45,19 +54,61 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 func runInit(dockerCli *client.DockerCli, flags *pflag.FlagSet, opts initOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
+	spec := opts.swarmOptions.ToSpec()
+
+	var secret string
+	// If no secret was specified, we create a random one
+	if !flags.Changed("secret") {
+		var secretBytes [generatedSecretEntropyBytes]byte
+
+		if _, err := cryptorand.Read(secretBytes[:]); err != nil {
+			panic(fmt.Errorf("failed to read random bytes: %v", err))
+		}
+
+		var nn big.Int
+		nn.SetBytes(secretBytes[:])
+		secret = fmt.Sprintf("%0[1]*s", maxGeneratedSecretLength, nn.Text(generatedSecretBase))
```

Comment:
This block (62-70) looks like it should be in its own function `func generateRandomSecret() string`